tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+  - project: /application/MQTT_AWS.cproject.yml
+  - project: /bootloader/Bootloader.cproject.yml
+    not-for-type: +Virtual
+```
+
+# Project Structure
+
+## Directory Structure  (old)
+
+This section describes how the files of `*.csolution.yml` should be organized to allow the scenarios described above:
+
+Source Directory                  | Content
+:-----------------------------------|:---------------
+`.`                                 | Contains one or more `*.csolution.yml` files that describes an overall application.
+`./<project>`                       | Each project has its own directory
+`./<project>/RTE+<target>`          | Configurable files that are specific to a target have a specific directory.
+`./<project>/RTE`                   | Configurable files that are common to all targets may have a common directory.
+`./<project>/Layer+<target>/<name>` | `*.clayer.yml` and related source files of a layers that are specific to a target have a specific directory.
+`./<project>/Layer/<name>`          | `*.clayer.yml` and related source files of a layers that are common to all targets may have a common directory.
+
+**Notes:**
+
+- `./<project>/RTE+<target>` contains the *.cprj file that is generated by `projmgr`
+- Directory names `RTE` and `Layer` should become configurable.  ToDo: analyze impact.
+
+The `./RTE` directory structure is maintained by tools and has the following structure. You should not modify the structure of this directory.
+
+`RTE[+<target>]` Directory        | Content
+:---------------------------------|:---------------
+`.../.<build-type>`               | Contains the file `RTE_Components.h` that is specific to a `build-type`.
+`.../<component class>`           | Configurable files for each component class are stored in sub-folders. The name of this sub-folder is derived from the component class name.
+`.../<component class>/<Dname>`   | Configurable files of the component class that are device specific. It is generated when a component has a condition with a `Dname` attribute. (strictly speaking no longer needed, backward compatiblity to MDK?)
+`.../Device/<Dname>`              | Configurable files of the component class Device. This should have always a condition with a `Dname` attribute.
+
+Output Directory                             | Content
+:--------------------------------------------|:---------------
+`./<project>/Output+<target>`                | Contains the final binary and symbol files of a project. Each `build-type` shares the same output directory.
+`./<project>/.Intrim+<target>/.<build-type>` | Contains interim files (`*.o`, `*.lst`) fore each `build-type`
+
+**Note:**
+- The content of the `Output` directory is generated by the `CBuild` step.
+
+## Software Components
+
+Software components are re-usable library or source files that require no modification in the user application. Optionally, configurable source and header files are provided that allow to set parameters for the software component. 
```

Comment:
Yes components are re-usable and users want to re-use them. But many users may want to modify them anyway for trying out a patch, adding some debug to better understand the component, or because they want to fork from it...  so the assumption that the components "require no modification in the user application" looks possibly too restricting to me.  For each component, user may want to use it as read-only (as reference from CMSIS PACK ROOT), or use it as a local copy in his project.

Reply:
I agree this is not a good idea to modify all the reference libraries, but it's up to the end user to decide and we can't prevent him. When you keep it in CMSIS PACK ROOT, users will anyway remove the read-only properties of files and will impact all their projects instead of impacting only one.

As I see it this requirement is already listed in https://github.com/Open-CMSIS-Pack/devtools/pull/77 as "add operating mode where component files get copied from packs into the project tree (note: requires to explicitly list all header files that belong to a component)".

Reply:
I believe we need a concept for making a component "private" for further modification.  Would you be willing to work on that?  What needs to happen in the file tree and the *.yml files in such a case?

Reply:
I'll try to work with colleagues on a possible proposal for this